colmena/models.py

```diff
@@ -163,17 +168,43 @@ def serialize(self) -> float:
         start_time = perf_counter()
         _value = self.value
         _inputs = self.inputs
+
+        def _serialize_and_proxy(value):
+            """Helper function for serializing and proxying"""
+            # Serialized object before proxying to compare size of serialized
+            # object to value server threshold
+            value_str = SerializationMethod.serialize(
+                self.serialization_method, value
+            )
+
+            if (
+                self.value_server_threshold is not None and
+                sys.getsizeof(value_str) >= self.value_server_threshold and
+                not isinstance(value, ps.proxy.Proxy)
+            ):
+                # Proxy the value. Note: we use the id of the object as the key
+                # so calling to_proxy() on the same object multiple times
+                # does not create multiple copies in the value server.
+                value_proxy = ps.to_proxy(value, key=str(id(value)))
```

Comment:
Wanted to note here that since ProxyStore uses its own serialization strategy (modeled after FuncX's), we cannot pass `value_str` to `to_proxy()` so the result is that `to_proxy()` serializes `value` again.

This a bit inefficient. I am thinking about adding to ProxyStore a way to register a custom serialization function with the proxy so we can use Colmena's serialization utilities (and avoid the double serialization here).

Reply:
Makes sense. Sounds like a good optimization to perform later. 

Do you have an issue or a TODO comment about this?

Reply:
I opened one here: https://github.com/gpauloski/ProxyStore/issues/6

I need to spend some more time thinking about how to implement it. Another alternative is we just make a custom Factory class for Colmena (rather than using the default RedisFactory in ProxyStore), so that the custom Colmena Factory uses the Colmena serialization utilities.